sql-plugin/src/main/scala/com/nvidia/spark/rapids/SpillableColumnarBatch.scala

```diff
@@ -82,6 +85,11 @@ class SpillableColumnarBatchImpl (id: TempSpillBufferId,
    */
   def spillId: TempSpillBufferId = id
 
+  override lazy val sizeInBytes: Long =
+    withResource(RapidsBufferCatalog.acquireBuffer(id)) { buff =>
```

Comment:
Note it is not necessary to go through the buffer acquistion logic here.  The buffer is immutable in the catalog, and since we're not trying to access the actual data but only the metadata we don't need to acquire.  Might be worth adding a `RapidsBufferCatalog.getBufferSize(id)` method but not necessary as part of this PR.

Reply:
That was a larger change than I really wanted to do, which is why I left it this way. Yes it is just meta data, but where the meta data is stored is in the RapidsBuffer. If I remember correctly that was because we initially thought about compressing the data when we spilled to host memory or disk. I didn't want to make a design change without verifying that it would be an OK thing to do.

Reply:
Yes we originally thought we could mutate the buffer silently behind the scenes, but we've since moved to a setup where the `RapidsBufferId` describes an immutable piece of data.  That means compressing the data requires re-registering the buffer with a new buffer ID (and corresponding metadata).  We leaned towards this route because there were cases during shuffle where it would be problematic and surprising if a buffer contents mutated.

I'll file a followup to add a buffer size accessor to the catalog that doesn't require acquiring the buffer.